wallet/udb/vsp.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package udb
+
+import (
+	"decred.org/dcrwallet/wallet/walletdb"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+)
+
+var (
+	vspBucketKey = []byte("vsp")
+)
+
+type VSPTicket struct {
+	FeeHash chainhash.Hash
+	// TODO add vote preferences?
+	// VotingPreference map[string]string
+}
+
+// SetVSPTicket sets a VSPTicket record into the db.
+func SetVSPTicket(dbtx walletdb.ReadWriteTx, ticketHash *chainhash.Hash, record *VSPTicket) error {
+	bucket := dbtx.ReadWriteBucket(vspBucketKey)
+	serializedRecord := serializeVSPTicket(record)
+
+	// Save the creation date of the store.
+	return bucket.Put(ticketHash.CloneBytes(), serializedRecord)
+}
+
+// VSPTickets gets all vsp tickets from the db.
+func VSPTickets(dbtx walletdb.ReadTx) ([]*VSPTicket, error) {
```

Comment:
this function doesn't seem too useful as it only gets the database values, not the keys (the ticket hashes) which are associated to each ticket.